nixos/modules/services/networking/opsdroid.nix

```diff
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (lib) mkEnableOption mkPackageOptionMD mkOption;
+  cfg = config.services.opsdroid;
+  yaml = pkgs.formats.yaml { };
+  settingsFile = yaml.generate "opsdroid.yaml" cfg.settings;
+in
+{
+  meta.maintainers = [ lib.maintainers.raitobezarius ];
+  options.services.opsdroid = {
+    enable = mkEnableOption "opsdroid, bot framework";
+    package = mkPackageOptionMD pkgs "opsdroid" { };
+    secretsFile = mkOption {
+      type = types.path;
+      description = lib.mdDoc ''
+        A file containing multiple environment variable
+        declarations that contains secrets which will
+        be substituted in the YAML configuration file.
+      '';
+    };
+
+    settings = mkOption {
+      type = yaml.type;
+      default = {};
+      description = lib.mdDoc ''
+        Configuration for the Opsdroid bot framework.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.opsdroid = {
+      description = "opsdroid - a general chat bot framework written in Python";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      preStart = ''
+        umask 0077
```

Comment:
I wonder, whether setting the umask on `serviceConfig` would yield the same result.